turbine/src/cluster_nodes.rs

```diff
@@ -395,42 +401,39 @@ fn cmp_nodes_stake(a: &Node, b: &Node) -> Ordering {
 // same TVU socket-addr, we only send shreds to one of them.
 // Additionally limits number of nodes at the same IP address to
 // MAX_NUM_NODES_PER_IP_ADDRESS.
-fn dedup_tvu_addrs(nodes: &mut Vec<Node>) {
+fn dedup_tvu_addrs(nodes: &mut [Node]) {
     const TVU_PROTOCOLS: [Protocol; 2] = [Protocol::UDP, Protocol::QUIC];
     let capacity = nodes.len().saturating_mul(2);
     // Tracks (Protocol, SocketAddr) tuples already observed.
     let mut addrs = HashSet::with_capacity(capacity);
     // Maps IP addresses to number of nodes at that IP address.
     let mut counts = HashMap::with_capacity(capacity);
-    nodes.retain_mut(|node| {
```

Comment:
this used to guarantee a valid TVU address for everything in nodes. Do all of the consumers of nodes have protections to re-check validity?

Side note: your new function actually fits the function name better. I would call the old behavior _sanitizing_ tvu addrs

Reply:
It did not guarantee this, as staked nodes could pass this while not having a valid tvu address.

Reply:
you're right. Valid was way too strong a word. It made sure nodes were either staked or had a TVU address.

I guess the fact that the staked nodes need not have contact info or TVU address addresses my concern.